docs/reference/developer/dependencies.rst

```diff
@@ -1,19 +1,40 @@
 Dependencies
 ============
 
-Places where dependencies are specified:
+Version pinning
+###############
 
-  - Conda package build (`conda/meta.yml <https://github.com/scipp/scipp/blob/main/conda/meta.yaml>`_)
-  - Conan file (inline in `CMakeLists.txt <https://github.com/scipp/scipp/blob/main/lib/CMakeLists.txt>`_)
+Frequent issues such broken builds or inability to make a release have led us to freeze as many of our build dependencies as possible.
+Note that this does *not* imply that runtime dependencies of the final packages are frozen.
+There are a number of non-obvious dependencies that have led to surpising issues, so we attempt to document this below.
 
-Conda packages
-##############
+Places that define dependencies
+###############################
 
-Required conda packages to reproduce the build environment are fully specified in `conda/meta.yaml <https://github.com/scipp/scipp/blob/main/conda/meta.yaml>`_
-These include runtime dependencies for scipp.
-  
-Conan
-#####
-
-Required conan dependencies are fully specified inline in `CMakeLists.txt <https://github.com/scipp/scipp/blob/main/lib/CMakeLists.txt>`_
-Note that these are brought into the project at cmake-time for the purposes of building and testing scipp.
+- GitHub workflows define ``runs-on``, i.e., VM image versions.
+  We are not using the ``*-latest`` images but instead rely on specific ones such as ``ubuntu-20.04`` or ``winddows-2019``.
+  Unfortunately GitHub appears to make changes even to the non-latest images, e.g., we once experienced build breaks from a compiler update in the ``windows-2019`` images.
+  In other words, the VM image "versions" are not truly frozen.
+- Actions for GitHub workflows are frozen to specific versions.
+  Dependabot is configured to update these regularely.
+  Note one known issue:
+  ``pypa/cibuildwheel`` updates may add support for new Python versions, but our ``pybind11`` may not be supporting it.
+  Make sure to run the ``release.yml`` workflow before accepting an update, or consider adding future Python versions to the ``[tool.cibuildwheel]`` ``skip`` list to make support for new Python versions a process fully under our control.
+- `pyproject.toml <https://github.com/scipp/scipp/blob/main/pyproject.toml>`_ defines build requirements for Python Wheels.
+- `setup.py <https://github.com/scipp/scipp/blob/main/setup.py>`_ defines runtime requirements and "extras" for the ``pip`` package.
+- All Python dependencies for static analysis, tests, documentation, and more are defined in ``requirements/*.in``.
+  ``pip-compile-multi`` is used to generate the corresponding ``requirements/*.txt`` files.
+  This should be run regularly, e.g., after a release.
```

Comment:
```suggestion
  This should be run regularly, e.g., before a release.
```
?

Reply:
No, because updating may bring issues, and will need time to fix. We therefor want to do it *after* releases.

Reply:
But users might have the latest version and thus install a broken package.

Reply:
This will also happen within a release cycle, we just need to make patch releases for this.